mobile/library/kotlin/io/envoyproxy/envoymobile/EngineBuilder.kt

```diff
@@ -571,6 +571,27 @@ open class EngineBuilder(
     return this
   }
 
+  /**
+   * Add a google_grpc RTDS layer to config
+   *
+   * @param layer_name the layer name.
+   *
+   * @param api_type sotw vs delta typw.
+   *
+   * @param address the GRPC server address.
+   *
+   * @param port the GRPC server port.
+   *
+   * @return this builder.
+   */
+  fun addRtdsLayer(layer_name: String, api_type: String, address: String, port: Int): EngineBuilder {
+    this.rtds_layer_name_ = layer_name
+    this.rtds_api_type_ = api_type
+    this.rtds_address_ = address
+    this.rtds_port_ = port
+    return this
+  }
```

Comment:
what is this doing exactly? the Kotlin functions would need to use a bridge to call the C++ APIs, no?  for starters, i think it needs to be passed into https://sourcegraph.com/github.com/envoyproxy/envoy@ec9099786796da6f834a6d562d0c3939c342a5e1/-/blob/mobile/library/kotlin/io/envoyproxy/envoymobile/EngineBuilder.kt?L583, which would then go through some other layers till it's initializing the Engine creation.

Also, I don't think you declared the rtds_* variables in the Kotlin file, so this probably won't compile.